src/components/Courses/Courses.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+
+import { useState } from 'react';
+
+import Button from 'src/common/Button/Button';
+
+import { getCourseById } from 'src/helpers/getCourseById';
+
+import CourseCard from './components/CourseCard/CourseCard';
+import SearchBar from './components/SearchBar/SearchBar';
+import EmptyCourseList from '../EmptyCourseList/EmptyCourseList';
+import CourseInfo from '../CourseInfo/CourseInfo';
+
+import { ADD_COURSE_BTN } from 'src/constants';
+
+interface coursesProp {
+	data: object[];
+}
+
+export default function Courses(props: coursesProp) {
+	const { data } = props;
+
+	const [courses, setCourses] = useState(data);
+	const [showInfo, setShowInfo] = useState(false);
+	const [courseId, setCourseId] = useState('');
+	const components =
+		courses.length !== 0 ? (
+			courses.map((course, id) => {
+				return (
+					<CourseCard
+						course={course}
+						setShowInfo={setShowInfo}
+						setCourseId={setCourseId}
+						key={id}
+					/>
+				);
+			})
+		) : (
+			<></>
+		);
+	const btn = <Button text={ADD_COURSE_BTN} />;
+	const empty = <EmptyCourseList />;
+	const list = <div>{components}</div>;
+	const view = <>{courses.length !== 0 ? list : empty}</>;
+	const component = showInfo ? (
+		<CourseInfo
+			course={getCourseById(courseId)}
+			setShowInfo={setShowInfo}
```

Comment:
Let's pass `onBackClick` or `onBackButton` click prop to `CourseInfo` and call it there when Back button is clicked.

```suggestion
			onBackClick={() => setShowInfo(false)}
```

Reply:
Thanks. Good idea, I agree to use it but I would like to discuss hooks(useEffect) so for now I used useEffect here. I am not sure that this is the best idea. Please take a look on it.